paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc

```diff
@@ -243,7 +275,7 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
 
         auto user_residual_md = platform::MKLDNNMemDesc(
             residual_data_tz, residual_data_type, residual_param->format());
-        user_residual_memory_p = handler.AcquireResidualDataMemory(
+        auto user_residual_memory_p = handler.AcquireResidualDataMemory(
```

Comment:
Please do not declare user_residual_memory_p in that location/scope. Actual mkl-dnn pipeline execution is in parent scope. It may happen that user_residual_memory_p is released while reorders/conv prim needs to access it. This may(and will) result in crash (@LeoZhao-Intel  encountered crash due to that problem)